### PR TITLE
FIX: SQLi in poller's broker configuration 21.04.x

### PR DESCRIPTION
**Pull request by hyahiaoui-ext**
## Description

Sanitizing pollers broker config queries.

**Fixes** # MON-14981

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by dmyios — APPROVED** [review]

File: www/include/configuration/configCentreonBroker/DB-Func.php [hunk 1]
```diff
@@ -104,8 +108,10 @@ function deleteCentreonBrokerInDB($ids = array())
 {
     global $pearDB;
 
+    $statement = $pearDB->prepare("DELETE FROM cfg_centreonbroker WHERE config_id = :config_id");
     foreach ($ids as $key => $value) {
-        $pearDB->query("DELETE FROM cfg_centreonbroker WHERE config_id = " . $key);
+        $statement->bindValue(':config_id', (int) $key, \PDO::PARAM_INT);
+        $statement->execute();
     }
```
Comment (TamazC) [inline www/include/configuration/configCentreonBroker/DB-Func.php:115]: `
function deleteCentreonBrokerInDB($ids = [])
{
    global $pearDB;

    $configIds = array_map(function($key) {
        return (int) $key;
    }, array_keys($ids));

    $statement = $pearDB->prepare('DELETE FROM cfg_centreonbroker WHERE config_id IN (:configIds)');
    $statement->bindValue(':config_id', join(',', $configIds), \PDO::PARAM_STMT);
    return $statement->execute();
}
`

is more elegant, you dont need to loop the delete statement

File: www/include/configuration/configCentreonBroker/DB-Func.php [hunk 2]
```diff
@@ -293,3 +291,54 @@ function isPositiveNumeric($size): bool
...
+             . "WHERE config_id = :config_id ";
+    try {
+        $statement = $pearDB->prepare($query);
+        $statement->bindValue(':config_id', $configId, \PDO::PARAM_INT);
+        $statement->execute();
+        $cfgBrokerData = $statement->fetch(\PDO::FETCH_ASSOC);
+    } catch (PDOException $exception) {
+        throw new \Exception("Cannot fetch Broker config data");
+    }
+    $statement->closeCursor();
```
Comment (TamazC) [inline www/include/configuration/configCentreonBroker/DB-Func.php:317]:
```suggestion
```

Comment (TamazC) [reply www/include/configuration/configCentreonBroker/DB-Func.php:317]: No need to close the close the cursor. As far as stateletn is explicitly unset or goes out of scope, closing will be performed automaticaly.

File: www/include/configuration/configCentreonBroker/DB-Func.php [hunk 3]
```diff
@@ -293,3 +291,54 @@ function isPositiveNumeric($size): bool
...
+             . "WHERE config_id = :config_id";
+    try {
+        $statement = $pearDB->prepare($query);
+        $statement->bindValue(':config_id', $configId, \PDO::PARAM_INT);
+        $statement->execute();
+        $cfgBrokerInfoData = $statement->fetchAll(\PDO::FETCH_ASSOC);
+    } catch (\PDOException $exception) {
+        throw new \Exception("Cannot fetch Broker info config data");
+    }
+    $statement->closeCursor();
```
Comment (TamazC) [inline www/include/configuration/configCentreonBroker/DB-Func.php:342]: same here

File: www/include/configuration/configCentreonBroker/DB-Func.php [hunk 4]
```diff
@@ -194,13 +200,7 @@ function multipleCentreonBrokerInDB($ids, $nbrDup)
...
         $cbObj = new CentreonConfigCentreonBroker($pearDB);
 
-        $query = "SELECT config_name, config_filename, config_activate, ns_nagios_server,
-            event_queue_max_size, cache_directory, daemon "
-            . "FROM cfg_centreonbroker "
-            . "WHERE config_id = " . $id . " ";
-        $dbResult = $pearDB->query($query);
-        $row = $dbResult->fetch();
-        $dbResult->closeCursor();
+        $row = getCfgBrokerData((int) $id);
```
Comment (TamazC) [inline www/include/configuration/configCentreonBroker/DB-Func.php:203]: 👍

File: www/include/configuration/configCentreonBroker/DB-Func.php [hunk 5]
```diff
@@ -210,14 +210,11 @@ function multipleCentreonBrokerInDB($ids, $nbrDup)
...
         $values['activate_watchdog']['activate_watchdog'] = $row['daemon'];
-        $query = "SELECT config_key, config_value, config_group, config_group_id "
-            . "FROM cfg_centreonbroker_info "
-            . "WHERE config_id = " . $id . " ";
-        $dbResult = $pearDB->query($query);
         $values['output'] = array();
         $values['input'] = array();
         $values['logger'] = array();
-        while ($rowOpt = $dbResult->fetch()) {
+        $brokerCfgInfoData = getCfgBrokerInfoData((int) $id);
```
Comment (TamazC) [inline www/include/configuration/configCentreonBroker/DB-Func.php:216]: 👍

File: www/include/configuration/configCentreonBroker/DB-Func.php (hunk 1 above)
Comment (hyahiaoui-ext) [reply www/include/configuration/configCentreonBroker/DB-Func.php:115]: These are cool, too bad these PRs are only backports, i wouldv made them in the original, maybe next one :) !

**Pull request merged**